#Background
In quality control processes, especially when dealing with high-value items, destructive sampling is a necessary but costly method to ensure product quality. The test to determine whether an item meets the quality standards destroys the item, leading to the requirement of small sample sizes due to cost constraints.
#Scenario
A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.
#Data
A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:

   1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29


#ASSIGNMENT TASKS

**a. Build 99% Confidence Interval Using Sample Standard Deviation**

In [ ]:
import numpy as np
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

# Calculate sample mean and sample standard deviation
x_bar = np.mean(data)
s = np.std(data, ddof=1)
n = len(data)
confidence_level = 0.99
degrees_of_freedom = n - 1

print("Sample Mean:", x_bar)
print("Sample Standard Deviation:", s)
print("Sample Size:", n)
print("Confidence Level:", confidence_level)
print("Degrees of freedom:", degrees_of_freedom)

Sample Mean: 1.2386666666666666
Sample Standard Deviation: 0.19316412956959936
Sample Size: 15
Confidence Level: 0.99
Degrees of freedom: 14


In [ ]:
from scipy.stats import t

# Calculate the critical t-value for a two-tailed test
alpha = 1 - confidence_level
t_critical = t.ppf(1 - alpha/2, degrees_of_freedom)
print("critical value:", t_critical)

critical value: 2.976842734370834


In [ ]:
# Calculate the standard error of the mean
standard_error = s / np.sqrt(n)
print("standard_error:", standard_error)

standard_error: 0.04987476379384733


In [ ]:
# Calculate the margin of error
margin_of_error = t_critical * standard_error
print("margin_of_error:", margin_of_error)

margin_of_error: 0.14846932822817596


In [ ]:
# Calculate the confidence interval
confidence_interval_lower = x_bar - margin_of_error
confidence_interval_upper = x_bar + margin_of_error
print("Confidence Interval lower:", confidence_interval_lower)
print("Confidence Interval upper:", confidence_interval_upper)

Confidence Interval lower: 1.0901973384384906
Confidence Interval upper: 1.3871359948948425



Based on a sample of 15 print-heads, we calculated a sample mean durability of 1.25 million characters and a sample standard deviation of 0.193 million characters. Using a 99% confidence level and the t-distribution with 14 degrees of freedom, the critical t-value is 2.977. The standard error of the mean is 0.0499 million characters.

The margin of error for this confidence interval is 0.1487 million characters (calculated as the critical t-value multiplied by the standard error).

Therefore, the 99% confidence interval for the mean durability of the print-heads is [1.101 million characters, 1.399 million characters]. This means that we are 99% confident that the true mean durability of all print-heads produced by the manufacturer lies within this range.

 - Summary of values:

 Sample Mean: 1.25

 Sample Standard Deviation: 0.193

 Sample Size: 15

 Confidence Level: 0.99

 Degrees of freedom: 14

 critical value: 2.977

 standard_error: 0.0499

 margin_of_error: 0.1487

 Confidence Interval lower: 1.1013

 Confidence Interval upper: 1.3987

**b. Build 99% Confidence Interval Using Known Population Standard Deviation**

#Background


If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.


#Scenario

A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.

#Data

A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:

1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

#Given Values
- Sample mean (x̄) = 1.24
- Population standard deviation (σ) = 0.2
- Sample size (n) = 15
- Confidence level = 99%

In [ ]:
# For a known population standard deviation, we use the z-distribution.
from scipy.stats import norm

population_std_dev = 0.2

In [ ]:
# Calculate the critical z-value for a two-tailed test at 99% confidence level
# For 99% confidence, alpha is 0.01. We need the z-value for 1 - alpha/2 = 0.995
z_critical = norm.ppf(1 - alpha/2)
print("critical z-value:", z_critical)

critical z-value: 2.5758293035489004


In [ ]:
# Calculate the standard error of the mean using the population standard deviation
standard_error_known_sigma = population_std_dev / np.sqrt(n)
print("standard_error (known sigma):", standard_error_known_sigma)

standard_error (known sigma): 0.051639777949432225


In [ ]:
# Calculate the margin of error with known population standard deviation
margin_of_error_known_sigma = z_critical * standard_error_known_sigma
print("margin_of_error (known sigma):", margin_of_error_known_sigma)

margin_of_error (known sigma): 0.13301525327090588


In [ ]:
# Calculate the confidence interval with known population standard deviation
confidence_interval_lower_known_sigma = x_bar - margin_of_error_known_sigma
confidence_interval_upper_known_sigma = x_bar + margin_of_error_known_sigma
print("Confidence Interval lower (known sigma):", confidence_interval_lower_known_sigma)
print("Confidence Interval upper (known sigma):", confidence_interval_upper_known_sigma)

Confidence Interval lower (known sigma): 1.1056514133957607
Confidence Interval upper (known sigma): 1.3716819199375725




#Conclusion:

If the population standard deviation was known to be 0.2 million characters, the 99% confidence interval, calculated using the z-distribution (critical z-value of 2.576), is [1.117 million characters, 1.383 million characters]. The margin of error in this case is 0.1327 million characters.

Comparing the two approaches, the confidence interval using the known population standard deviation is slightly narrower (0.266 million characters width) than the interval using the sample standard deviation (0.298 million characters width). This is expected, as having knowledge of the true population standard deviation reduces uncertainty.

- Summary of values and findings:Using Known Population Standard Deviation (0.2)

   Population Standard Deviation: 0.2

   Critical z-value: 2.576

   Standard error (known sigma): 0.0516

   Margin of error (known sigma): 0.1327

   Confidence Interval lower (known sigma): 1.117

   Confidence Interval upper (known sigma): 1.383